src/core/csv/reader_parsers.cc

```diff
@@ -147,7 +148,8 @@ void parse_int_simple(FreadTokenizer& ctx) {
       (sd == MAX_DIGITS && value <= MAX_VALUE))
   {
     T x = static_cast<T>(value);
-    *reinterpret_cast<T*>(ctx.target) = (x ^ -negative) + negative;
+    *reinterpret_cast<T*>(ctx.target) = (x ^ -static_cast<int>(negative))
+                                        + negative;
```

Comment:
if `x` is of type `T`, then shouldn't `negative` be cast to the same type?
```
T neg = static_cast<T>(negative);
*reinterpret_cast<T*>(ctx.target) = (x ^ -neg) + neg;  // same as neg? -x : x
```

Reply:
I guess I casted it to `int` because `T` could only be `int32_t` or `int64_t`, we can of course cast it to `T` for flexibility and clarity.